### PR TITLE
[34633] Add check for PO close when all items closed

### PR DESCRIPTION
**Pull request by andersdd**

### CODE REVIEW

File: guiclient/purchaseOrder.cpp [hunk 1]
```diff
@@ -852,6 +852,29 @@ void purchaseOrder::sSave()
...
+                        "  AND poitem_status <> 'C';" );
+  purchaseSave.bindValue(":pohead_id", _poheadid);
+  purchaseSave.exec();
+  if (purchaseSave.first() && purchaseSave.value("isOpen").toInt() == 0)
+  {
+     if (QMessageBox::question(this, tr("Close Purchase Order?"),
+                                  tr("<p>There are no open items on this Purchase Order. "
+                                     "Do you wish to close this Purchase Order?"),
+         QMessageBox::Yes,
+          QMessageBox::No | QMessageBox::Default) == QMessageBox::Yes)
```
Comment (gilmoskowitz) [inline guiclient/purchaseOrder.cpp]: This is a deprecated form of `QMessageBox::question()`.

File: guiclient/purchaseOrder.cpp [hunk 2]
```diff
@@ -852,6 +852,29 @@ void purchaseOrder::sSave()
   if (GuiErrorCheck::reportErrors(this, tr("Cannot Save Purchase Order"), errors))
     return;
 
+  purchaseSave.prepare( "SELECT count(*) AS isOpen "
+                        "FROM poitem "
+                        "WHERE poitem_pohead_id=:pohead_id "
+                        "  AND poitem_status <> 'C';" );
```
Comment (gilmoskowitz) [inline guiclient/purchaseOrder.cpp]: Watch coding standard here:
- [indentation](https://github.com/xtuple/qt-client/wiki/xTuple-Coding-Standards#sql-and-metasql)
- Consider [SELECT EXISTS(SELECT 1 ...)](https://github.com/xtuple/qt-client/wiki/xTuple-Coding-Standards#plpgsql) instead

File: guiclient/purchaseOrder.cpp [hunk 3]
```diff
@@ -852,6 +852,29 @@ void purchaseOrder::sSave()
...
+  purchaseSave.exec();
+  if (purchaseSave.first() && purchaseSave.value("isOpen").toInt() == 0)
+  {
+     if (QMessageBox::question(this, tr("Close Purchase Order?"),
+                                  tr("<p>There are no open items on this Purchase Order. "
+                                     "Do you wish to close this Purchase Order?"),
+         QMessageBox::Yes,
+          QMessageBox::No | QMessageBox::Default) == QMessageBox::Yes)
+     {
+       _status->setCurrentIndex(2);
```
Comment (gilmoskowitz) [inline guiclient/purchaseOrder.cpp]: This line works the same way as the rest of this file handles `_status`. OTOH we're trying to replace that wherever we can with symbolic manipulation of the widget. Examples:
- [dspGLSeries](https://github.com/xtuple/qt-client/blob/dbd44398d275e9550f53b5b3764b18d06b361666/guiclient/displays/dspGLSeries.cpp#L47-L56)
- [returnAuthorization](https://github.com/xtuple/qt-client/blob/4_12_x/guiclient/returnAuthorization.cpp#L47-L59)
- [salesOrder](https://github.com/xtuple/qt-client/blob/4_12_x/guiclient/salesOrder.cpp#L163-L169)

File: guiclient/purchaseOrder.cpp (hunk 1 above)
Comment (andersdd) [reply guiclient/purchaseOrder.cpp]: Updated - I will get it right one year.

File: guiclient/purchaseOrder.cpp (hunk 2 above)
Comment (andersdd) [reply guiclient/purchaseOrder.cpp]: Updated SQL

File: guiclient/purchaseOrder.cpp (hunk 3 above)
Comment (andersdd) [reply guiclient/purchaseOrder.cpp]: Modified Purchase Order status widget to use codes.

**Pull request merged**